pkg/minikube/machine/cache_images.go

```diff
@@ -770,6 +763,42 @@ func ListImages(profile *config.Profile, format string) error {
 	return nil
 }
 
+// mergeImageLists merges image lists from different nodes into a single list
+// all the repo tags of the same image will be preserved and grouped in one image item
+func mergeImageLists(lists [][]cruntime.ListImage) []cruntime.ListImage {
+	images := map[string]*cruntime.ListImage{}
+	// key of this set is img.ID+img.repoTag
+	// we use this set to detect whether a tag of a image is included in images map
+	imageTagAndIDSet := map[string]struct{}{}
+	for _, list := range lists {
+		for _, img := range list {
+			img := img
+			if _, ok := images[img.ID]; !ok {
+				images[img.ID] = &img
+				for _, repoTag := range img.RepoTags {
+					imageTagAndIDSet[img.ID+repoTag] = struct{}{}
+				}
+				continue
+			}
+			for _, repoTag := range img.RepoTags {
+				if _, ok := imageTagAndIDSet[img.ID+repoTag]; ok {
+					continue
+				}
+				// if there is any repo tag which is not included in the map's corresponding item
+				// add it to the repo tag list
+				imageTagAndIDSet[img.ID+repoTag] = struct{}{}
+				images[img.ID].RepoTags = append(images[img.ID].RepoTags, repoTag)
```

Comment:
```suggestion
			_, existingImg := images[img.ID]
			if !existingImg {
				images[img.ID] = &img
			}
			for _, repoTag := range img.RepoTags {
			        if _, ok := imageTagAndIDSet[img.ID+repoTag]; ok {
					continue
				}
				imageTagAndIDSet[img.ID+repoTag] = struct{}{}
				if existingImg {
				        images[img.ID].RepoTags = append(images[img.ID].RepoTags, repoTag)
				}
```

Remove duplication and reduce nesting